src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/CSharpDiagnosticWorkerWithAnalyzers.cs

```diff
@@ -235,12 +235,12 @@ private async Task AnalyzeDocument(Project project, ImmutableArray<DiagnosticAna
                 }
                 else if (allAnalyzers.Any()) // Analyzers cannot be called with empty analyzer list.
                 {
-                    var semanticDiagnosticsWithAnalyzers = await compiled
-                        .WithAnalyzers(allAnalyzers, workspaceAnalyzerOptions)
+                    var compilationWithAnalyzers = compiled.WithAnalyzers(allAnalyzers, new CompilationWithAnalyzersOptions(workspaceAnalyzerOptions, null, false, false, false));
```

Comment:
Sorry for delay

It could be good to add named parameters since its much easier to understand how analyzers are configured with them, like reportSuppressedDiagnostics: false instead of "false" :thinking: 

About error logging, personally i feel that it could be good idea to pass logger function instead of null. Situations what i am thinking are mainly cases like missing services from workspaces etc because there truly are differences between VS and omnisharp in certain cases + configurations may differ which can trigger error situations. That way instead of issue "missing diagnostic X" becomes "missing diagnostic X and heres these errors on logs". Trace could be good place to start as severity since its common to ask trace level logs on issue and then real error is catch up.